scalafix/rules/src/main/scala/fix/v0_14_0/FixAvroCoder.scala

```diff
@@ -129,6 +134,32 @@ object FixAvroCoder {
 
   def isAvroType(sym: Symbol)(implicit sd: SemanticDocument): Boolean =
     AvroMatcher.matches(sym) || hasParentClass(sym, AvroMatcher)
+
+  def hasAvroTypeSignature(tree: Tree, checkLiftedType: Boolean)(implicit doc: SemanticDocument): Boolean =
+    tree.symbol.info.map(_.signature).exists {
+      case MethodSignature(_, _, TypeRef(_, returnType, _)) if !checkLiftedType && isAvroType(returnType) =>
+        true
+      case MethodSignature(_, _, TypeRef(_, _, List(TypeRef(_, returnType, _)))) if checkLiftedType && isAvroType(returnType) =>
+        true
+      case _ =>
+        false
+    }
+
+  def methodHasAvroCoderTypeBound(expr: Term)(implicit doc: SemanticDocument): Boolean =
+    expr.symbol.info.map(_.signature) match {
+      case Some(MethodSignature(_, parameterLists, _)) =>
+        parameterLists.flatten.flatMap(_.symbol.info.map(_.signature)).exists {
+          case ValueSignature(TypeRef(_, symbol, List(TypeRef(_, coderT, _)))) if CoderMatcher.matches(symbol) =>
+            coderT.info.map(_.signature).exists {
+              case TypeSignature(_, _, TypeRef(_, maybeAvroType, _)) if AvroMatcher.matches(maybeAvroType) =>
+                true
+              case _ =>
+                false
+            }
+          case _ => false
+        }
+      case _ => false
+    }
```

Comment:
Nit, you can remove nesting by converting the option to List and using collect

```suggestion
    expr.symbol.info.map(_.signature)
      .toList
      .collect { case MethodSignature(_, parameterLists, _) => parameterLists }
      .flatMap(_.flatten)
      .flatMap(_.symbol.info.map(_.signature).toList)
      .collect { case ValueSignature(TypeRef(_, symbol, List(TypeRef(_, coderT, _)))) if CoderMatcher.matches(symbol) => coderT }
      .flatMap(_.info.map(_.signature).toList)
      .exists { case TypeSignature(_, _, TypeRef(_, maybeAvroType, _)) => AvroMatcher.matches(maybeAvroType) }
```